Mathlib/RingTheory/Smooth/Kaehler.lean

```diff
@@ -224,3 +238,195 @@ def retractionKerToTensorEquivSection :
     simp only [this, Algebra.algebraMap_eq_smul_one, ← smul_tmul', LinearMapClass.map_smul,
       SetLike.val_smul, smul_eq_mul, sub_zero]
   right_inv g := by ext s; obtain ⟨s, rfl⟩ := hf s; simp
+
+variable (R P S) in
+/--
+Given a tower of algebras `S/P/R`, with `I = ker(P → S)`,
+this is the `R`-derivative `P/I² → S ⊗ Ω[P] Ω[P⁄R]` given by `[x] ↦ 1 ⊗ D x`.
+-/
+noncomputable
+def derivationQuotKerSq :
+    Derivation R (P ⧸ (RingHom.ker (algebraMap P S) ^ 2)) (S ⊗[P] Ω[P⁄R]) := by
+  letI := Submodule.liftQ ((RingHom.ker (algebraMap P S) ^ 2).restrictScalars R)
+    (((mk P S _ 1).restrictScalars R).comp (KaehlerDifferential.D R P).toLinearMap)
```

Comment:
I was told before to avoid tactic proofs when constructing data. Could you try to avoid it here (and below)?

Reply:
I think it is only necessary to avoid tactics that rewrites the types of the goal e.g. `rw` or `simp` etc.
I've changed `tensorKaehlerQuotKerSqEquiv` but changing the others would make things a lot more messy.